compiler/x/codegen/ControlFlowEvaluator.cpp

```diff
@@ -1312,6 +1316,24 @@ TR::Register *OMR::X86::TreeEvaluator::iternaryEvaluator(TR::Node *node, TR::Cod
       else
          generateRegRegInstruction(CMOVERegReg(trueValIs64Bit), node, trueReg, falseReg, cg);
       }
+   else if (!longCompareOn32bit && conditionOp.isCompareForOrder() && condition->getFirstChild()->getOpCode().isIntegerOrAddress())
+      {
+      TR::TreeEvaluator::compareIntegersForOrder(condition, cg);
+      if (conditionOp.isCompareTrueIfEqual())
+         {
+         if (conditionOp.isCompareTrueIfGreater())
+            generateRegRegInstruction(CMOVLRegReg(trueValIs64Bit), node, trueReg, falseReg, cg);
+         else
+            generateRegRegInstruction(CMOVGRegReg(trueValIs64Bit), node, trueReg, falseReg, cg);
+         }
+      else
+         {
+         if (conditionOp.isCompareTrueIfGreater())
+            generateRegRegInstruction(CMOVLERegReg(trueValIs64Bit), node, trueReg, falseReg, cg);
+         else
+            generateRegRegInstruction(CMOVGERegReg(trueValIs64Bit), node, trueReg, falseReg, cg);
+         }
+      }
```

Comment:
The above sequences of calls are all identical except for the opcode.  Can you change the if/else ladder above to just figure out the opcode, and then just do one call to `generateRegRegInstruction` with that opcode afterwards?  This will save some DLL space.

Reply:
Addressed.